src/SFA.DAS.ApprenticeAccounts/Application/Commands/CreateMyApprenticeCommand/CreateMyApprenticeshipCommandValidator.cs

```diff
@@ -31,14 +30,24 @@ public CreateMyApprenticeshipCommandValidator(IApprenticeContext apprenticeConte
                 return result != null;
             }).WithMessage(ApprenticeIdNotPresent);
 
-        RuleFor(model => model.ApprenticeshipId)
-            .Must((model,apprenticeshipId, cancellation) =>
-            { 
-                if (model.ApprenticeshipId == null) return true;
-        
-                var myApprenticeships =  myApprenticeshipContext.FindAll(model.ApprenticeId).Result;
-                return myApprenticeships.All(x => x.ApprenticeshipId != apprenticeshipId);
+        RuleFor(model => model.ApprenticeId)
+            .Must((model, cancellation) =>
+            {
+                var myApprenticeship =  myApprenticeshipContext.FindByApprenticeId(model.ApprenticeId).Result;
+                return myApprenticeship == null;
         
-            }).WithMessage(ApprenticeshipIdAlreadyExists);
+            }).WithMessage(MyApprenticeshipAlreadyPresent);
+
+        RuleFor(model => model.ApprenticeshipId)
+            .Must((model, cancellation) =>
+            {
+                if (model.ApprenticeshipId == null)
+                    return true;
+                var myApprenticeship =
+                    myApprenticeshipContext.FindByApprenticeshipId((long)model.ApprenticeshipId).Result;
+                
+                return myApprenticeship == null;
+
+            }).WithMessage(ApprenticeshipIdAlreadyPresent);
```

Comment:
This check should be in the handler and the handler should return null is there is no MyApprenticeship record which will result in NotFound response in the controller